src/modules/credits/sagas.spec.ts

```diff
@@ -0,0 +1,124 @@
+import { throwError } from 'redux-saga-test-plan/providers'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import { CONNECT_WALLET_SUCCESS } from 'decentraland-dapps/dist/modules/wallet/actions'
+import { Wallet } from 'decentraland-dapps/dist/modules/wallet/types'
+import { fetchCreditsRequest, fetchCreditsSuccess, fetchCreditsFailure, pollCreditsBalanceRequest } from './actions'
+import { creditsSaga, creditsService } from './sagas'
+import { getCredits } from './selectors'
+import { CreditsResponse } from './types'
+
+jest.mock('../vendor/decentraland/credits/api', () => ({
+  CreditsAPI: jest.fn().mockImplementation(() => ({
+    fetchCredits: jest.fn()
+  }))
+}))
+
+describe('Credits saga', () => {
+  const address = '0x123'
+  const mockCredits: CreditsResponse = {
+    totalCredits: 1000,
+    credits: [
+      {
+        id: '1',
+        amount: '1000',
+        availableAmount: '1000',
+        contract: '0x123',
+        expiresAt: '1000',
+        season: 1,
+        signature: '123',
+        timestamp: '1000',
+        userAddress: '0x123'
+      }
+    ]
+  }
+
+  beforeEach(() => {
+    jest.spyOn(creditsService, 'fetchCredits').mockResolvedValue(mockCredits)
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe('when handling fetchCreditsRequest action', () => {
+    it('should put fetchCreditsSuccess with the credits when the request succeeds', () => {
+      return expectSaga(creditsSaga)
+        .provide([[call([creditsService, 'fetchCredits'], address), mockCredits]])
+        .put(fetchCreditsSuccess(address, mockCredits))
+        .dispatch(fetchCreditsRequest(address))
+        .silentRun()
+    })
+
+    it('should put fetchCreditsFailure with the error message when the request fails with a message', () => {
+      const errorMessage = 'Invalid address'
+      const error = new Error(errorMessage)
+
+      return expectSaga(creditsSaga)
+        .provide([[call([creditsService, 'fetchCredits'], address), throwError(error)]])
+        .put(fetchCreditsFailure(address, errorMessage))
+        .dispatch(fetchCreditsRequest(address))
+        .silentRun()
+    })
+
+    it('should put fetchCreditsFailure with an unknown error when the request fails without a message', () => {
+      return expectSaga(creditsSaga)
+        .provide([[call([creditsService, 'fetchCredits'], address), throwError({} as Error)]])
+        .put(fetchCreditsFailure(address, t('global.unknown_error')))
+        .dispatch(fetchCreditsRequest(address))
+        .silentRun()
+    })
+  })
+
+  describe('when handling connectWalletSuccess action', () => {
+    it('should put fetchCreditsRequest with the wallet address', () => {
+      const wallet = { address } as Wallet
+
+      return expectSaga(creditsSaga)
+        .put(fetchCreditsRequest(address))
+        .dispatch({
+          type: CONNECT_WALLET_SUCCESS,
+          payload: { wallet }
+        })
```

Comment:
What do you think about using the `connectWalletSuccess` action creator here instead of the plain action?
```suggestion
        .dispatch({
          type: CONNECT_WALLET_SUCCESS,
          payload: { wallet }
        })
```